### ISSUE TITLE
Generates mutually exclusive condition from !String.IsNullOrEmpty(str) in LINQ.

### ISSUE THREAD

**Issue by xxlagr** [dotnet/efcore#12474]
**Condition**

`[e0].[Name] IS NOT NULL AND (([e0].[Name] <> N'') OR [e0].[Name] IS NULL)`

So value could be NULL and haven't be NULL simultaneously.

**Code**

```C#
public async Task<IEnumerable<Entry>> GetEntriesAsync() {
            var query = from e in context.Entries.Where(e => !String.IsNullOrEmpty(e.Name))
                        select new Entry
                        {
                           Id = e.Id,
                           Name = e.Name,
                        };
            return await query.AsNoTracking().ToListAsync();
}
```
**Result query**

```sql
SELECT [e0].[Id], [e0].[Name]
FROM [Project].[Entries] AS [e0]
WHERE ([e0].[Deleted] = 0) AND ([e0].[Name] IS NOT NULL AND (([e0].[Name] <> N'') OR [e0].[Name] IS NULL))
```

**Further technical details**
EF Core version: 2.1.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
IDE: Visual Studio 2017 15.7.4

**Comment by Sun5Kong** [comment]
the same in 2.1.11
will this problem be fix?

**Comment by ajcvickers** [comment]
@Sun5Kong This issue is in the Backlog milestone. This means that it is not going to happen for the 3.0 release. We will re-assess the backlog following the 3.0 release and consider this item at that time. However, keep in mind that there are many other high priority features with which it will be competing for resources.

**Comment by ranma42** [comment]
EFCore 5.0.17 emits
```sql
SELECT "b"."BlogId", "b"."NullableString"
FROM "Blogs" AS "b"
WHERE "b"."NullableString" IS NOT NULL AND (("b"."NullableString" <> '') OR "b"."NullableString" IS NULL)
```
but since 6.0.0 the query is simplified to
```sql
SELECT "b"."BlogId", "b"."NullableString"
FROM "Blogs" AS "b"
WHERE "b"."NullableString" IS NOT NULL AND ("b"."NullableString" <> '')
```

**Comment by roji** [comment]
Thanks @ranma42! Closing this as fixed in 6.0.0.
